### PR TITLE
[QA-981] Delete partially created groups that return 5xx when creating before retrying

### PR DESCRIPTION
**Pull request by dvoet**
**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [x] Get two thumbsworth of PR review
- [x] Verify all tests go green (CI _and_ coverage tests)
- [x] Squash commits and **merge to develop**
- [x] Delete branch after merge

### CODE REVIEW

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala [hunk 1]
```diff
@@ -107,9 +108,21 @@ class HttpGoogleDirectoryDAO(appName: String,
...
     for {
-      _ <- retry(when5xx, whenUsageLimited, when404, whenInvalidValueOnBucketCreation, whenNonHttpIOException)(() => {
+      _ <- retryWithRecover(when5xx,
+                            whenUsageLimited,
+                            when404,
+                            whenInvalidValueOnBucketCreation,
+                            whenNonHttpIOException)(() => {
         executeGoogleRequest(inserter)
-      })
+      }) {
```
Comment (gpolumbo-broad) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala]: nit of nits....would you mind indenting/formatting this block to align the parens and braces

**Review by gpolumbo-broad — APPROVED** [review]
Just a formatting nit, but functionally 👍

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala (hunk 1 above)
Comment (marctalbott) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala]: @gpolumbo-broad I'm not sure I understand which parens and braces you want to align. I formatted this using scalafmt, so this should be consistent with how we have formatted the rest of workbench-libs

Comment (marctalbott) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala]: I guess the stuff down on line 191-193 is similar, is that the indentation you'd like to see?

Comment (gpolumbo-broad) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala:117]: I just find it really hard to read.  the way it's formatted at the moment, I had to manually go one by one through the braces and parens to match them up to understand what block of code goes with what.  I appreciate that this is the way scalafmt wanted to format it, but with the currying on `retryWithRecover` compounded with the fact that we're already in a `for` comp, I could make sense of it, but I had to kind of parse this code in my head first just to be able to read it.  Like I said, I 👍 the PR, formatting isn't something I want to tie up PRs usually, but when indentation and formatting is done properly, you can sort of ignore braces and parens most of the time and it makes the code easier to read.  I guess my only request going forward is to use scalafmt to get us started, but if you think reformatting manually will make the code more readable and understandable, that's more important to me.

Comment (gpolumbo-broad) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala]: I just find it really hard to read.  the way it's formatted at the moment, I had to manually go one by one through the braces and parens to match them up to understand what block of code goes with what.  I appreciate that this is the way scalafmt wanted to format it, but with the currying on `retryWithRecover` compounded with the fact that we're already in a `for` comp, I could make sense of it, but I had to kind of parse this code in my head first just to be able to read it.  Like I said, I 👍 the PR, formatting isn't something I want to tie up PRs usually, but when indentation and formatting is done properly, you can sort of ignore braces and parens most of the time and it makes the code easier to read.  I guess my only request going forward is to use scalafmt to get us started, but if you think reformatting manually will make the code more readable and understandable, that's more important to me.

Comment (marctalbott) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala]: Sure, I'm happy to adjust the formatting and indentation to improve readability as it's an easy fix that can be really helpful in the future, I just wasn't sure what you had in mind. I switched it up a little, let me know if you think it's better.

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala [hunk 2]
```diff
@@ -107,9 +108,21 @@ class HttpGoogleDirectoryDAO(appName: String,
...
+        case t: Throwable if when5xx(t) =>
```
Comment (andy7i) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleDirectoryDAO.scala:118]: took me a bit to realize this is used within `retryWithRecover` (and not called after). oops

**Review by andy7i — APPROVED** [review]
LGTM 👍

**Pull request merged**